submissions/zalevskyi/dom/js/app.js

```diff
@@ -0,0 +1,110 @@
+        img_src: './img/Pingala.png',
+        img_alt: 'Pingala portrait'
+    },
+    'reyna' : {
+        type: 'governor',
+        title: 'The Financier',
+        description: 'Shrewd and fiscally knowledgeable, Reyna is able to strengthen the economic power of a city, using finances to provide more than is normally possible.',
+        context: 'A financier is an official tasked with resolving budgetary and financial matters, requiring an astute understanding of economics and the psychology of markets. Playing with the money of the state is not a matter for the faint-hearted.',
+        img_src: './img/Reyna.png',
+        img_alt: 'Reyna portrait'
+    },
+    'victor' : {
+        type: 'governor',
+        title: 'The Castellan',
+        description: 'Military focused, once established in a city, Victor is instrumental in strengthening, preserving, and defending it from foreign threats.',
+        context: 'A Castellan is a military officer responsible for directing the day-to-day affairs of a castle or fortress and the territory immediately around it. As castles were massive investments of treasure and important points of military power, the castellan had to be both competent and politically reliable. In times of war, a competent castellan could determine the fate of campaigns, and with that, the historical outcome.',
+        img_src: './img/Victor.png',
+        img_alt: 'Victor portrait'
+    }
+}
+const ARTICLE_INNER_HTML = {
+    'overview': `<h1 id='title' class='text_heading_1'></h1><p id='description' class='text'></p>`,
+    'governor': `<h1 id='title' class='text_heading_1'></h1>
+                 <img id='portrait' alt='' src='' class='portrait'>
+                 <h2 class='text_heading_2 description_heading'>Description</h2>
+                 <p id='description' class='text description'></p>
+                 <h2 class='text_heading_2 context_heading'>Historical Context</h2>
+                 <p id='context' class='text context'></p>`
+}
+Object.freeze(WIKI)
+Object.freeze(ARTICLE_INNER_HTML)
+function updateCommon(name) {
+    document.getElementById('title').textContent = `${name} - ${WIKI[name].title}`
+    document.getElementById('description').textContent = WIKI[name].description
+}
+function updateGovernor(name) {
+    document.getElementById('context').textContent = WIKI[name].context
+    document.getElementById('portrait').src = WIKI[name].img_src
+    document.getElementById('portrait').alt = WIKI[name].img_alt
+}
+document.getElementById('menu').addEventListener('click', ({target}) => {
+    const target_before = document.querySelector('.menu_item_selected')
+    if (target!==target_before) {
+        const article = document.getElementById('article')
+        const name = target.textContent
+        const name_before = target_before.textContent
+        target_before.classList.toggle('menu_item_selected')
+        target.classList.toggle('menu_item_selected')
+        article.classList.add('hidden')
```

Comment:
Why do you hide this element and then display it?

Reply:
Hiding and then showing again it was from
Udacity: JavaScript and the DOM: Lesson 4 Performance: 3 Reflow & Repaint [link](https://classroom.udacity.com/courses/ud117/lessons/364ca3da-1eb9-4213-beb4-7791ac4e1ea4/concepts/d1bee920-e33f-41db-aac3-2c3c5a08db3b)
> we could hide #comments, delete the spam, and show it again -- that's surprisingly fast, to the cost of one reflow and two repaints (and little else). It's fast because hiding doesn't change the layout, it just erases that section of the screen (1 repaint). When you make the changed section visible again, that's a reflow and a repaint.

Any way I have removed it and simply added image size into CSS to avoid reflow after image loading. For this case it works just fine.
Thank you for paying attention to it.

Reply:
ok, got it. It's up to you. let's leave it. no problem